tools/server/webui/src/utils/types.ts

```diff
@@ -113,6 +115,40 @@ export interface CanvasPyInterpreter {
 
 export type CanvasData = CanvasPyInterpreter;
 
+export interface ToolCallRequest {
+  id: string;
+  type: 'function';
+  call_id: string;
+  function: {
+    name: string;
+    arguments: string; // JSON string of arguments
+  };
+}
+
+export interface ToolCallSpec {
+  type: 'function';
+  function: {
+    name: string;
+    description: string;
+    parameters: ToolCallParameters;
+  };
+}
+export type ToolCallType = NonNullable<
+  (Message | PendingMessage)['tool_calls']
+>[number];
```

Comment:
Can we simply list the types here instead of having it inferred?

Like for example:

```
ToolCallType = 'function' | ...;
```

I don't think this type has many values for now, doing this way make the code more transparent

Reply:
Sorry! This datatype was just a redefined one. Fixed now.